Bottom: 43dda453f9f1db191fad4fbf43e9e1316d2aa76f
Top:    948a4c019ff549062edd463edf09dd778fd6fdbc
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-05-22 21:00:38 -0400

Refresh of mpc5200-ac97

---

diff --git a/sound/soc/fsl/mpc5200_psc_ac97.c b/sound/soc/fsl/mpc5200_psc_ac97.c
index 00e832f..ba8e114 100644
--- a/sound/soc/fsl/mpc5200_psc_ac97.c
+++ b/sound/soc/fsl/mpc5200_psc_ac97.c
@@ -322,6 +322,9 @@ static int __devinit psc_ac97_of_probe(struct of_device *op,
 	if (rc != 0)
 		return rc;
 
+	for (i = 0; i < ARRAY_SIZE(psc_ac97_dai); i++)
+		psc_ac97_dai[i].dev = &op->dev;
+
 	rc = snd_soc_register_dais(psc_ac97_dai, ARRAY_SIZE(psc_ac97_dai));
 	if (rc != 0) {
 		pr_err("Failed to register DAI\n");
